tests/pipeline/test_blaze.py

```diff
@@ -1645,6 +1751,164 @@ def test_deltas_before_index_0_macro(self):
+        # row (first in the group) has an earlier timestamp but later asof date
+        # so it should be selected.
+        deltas_df = pd.DataFrame({
+            'value': [
+                0.0,  # selected
+                1.0,  # ignored
+
+                2.0,  # ignored
+                3.0,  # selected
+
+                4.0,  # selected
+                5.0,  # ignored
+            ],
+            'asof_date': [
+                # swapped order: second row is before the first
+                T('2014-01-02'),
+                T('2014-01-01'),
+
+                # chronological order: second row is after the first
+                T('2014-01-03'),
+                T('2014-01-04'),
+
+                # swapped order: second row is before the first
+                T('2014-01-06'),
+                T('2014-01-05'),
+            ],
+            'timestamp': [
+                # we learn about all rows in monotonically increasing order
+                T('2013-01-02 22:00'),
+                T('2014-01-02 23:00'),
+                T('2014-01-04 22:00'),
+                T('2014-01-04 23:00'),
+                T('2014-01-06 22:00'),
+                T('2014-01-06 23:00'),
+            ],
+        })
+        deltas = bz.data(deltas_df, name='deltas', dshape=self.macro_dshape)
+
+        expected_views = keymap(pd.Timestamp, {
+            '2014-01-05': np.array([[0.0],
+                                    [3.0]]),
+            '2014-01-07': np.array([[3.0],
+                                    [4.0]]),
+        })
+
+        with tmp_asset_finder(equities=simple_asset_info) as finder:
+            # The dates queried are non-contiguous. We have two day groups to
+            # capture the two day pairs in the input data.
+            dates = pd.to_datetime(['2014-01-03', '2014-01-05', '2014-01-07'])
+            self._run_pipeline(
+                expr=expr,
+                deltas=deltas,
+                checkpoints=None,
+                expected_views=expected_views,
+                expected_output=None,
+                finder=finder,
+                calendar=dates,
+                start=dates[1],
+                end=dates[-1],
+                window_length=2,
+            )
+
+    def test_stacked_deltas_macro(self):
+        df = empty_dataframe(
+            ('value', 'float64'),
+            ('asof_date', 'datetime64[ns]'),
+            ('timestamp', 'datetime64[ns]'),
+        )
+        expr = bz.data(df, name='expr', dshape=self.macro_dshape)
+
+        T = pd.Timestamp
+
+        # These data are interesting because they exercise the tie breaking of
+        # adjustments. Here we have 4 rows which we learn about within a single
+        # calendar index. The first row provides the most recently known value
+        # for some day in the window. All of the following rows are adjustments
+        # to the same historical value. We expect that the first row's value is
```

Comment:
"an earlier historical value" might make it clearer that the adjustments are for a different date than the first value.

Reply:
"earlier historical" seems a bit redundant